### PR TITLE
Review tint, shade and tone in TileColor #87

### PR DESCRIPTION
**Pull request by klaushauschild1984**
implement some tests

This is my first attempt to do some Kotlin. I did some edge case and in-between testing, nothing fancy.

### CODE REVIEW

File: zircon.core/src/jvmTest/kotlin/org/hexworks/zircon/internal/color/DefaultTileColorTest.kt [hunk 1]
```diff
@@ -23,6 +24,73 @@ class DefaultTileColorTest {
         assertThat(result.alpha).isEqualTo(TileColor.defaultAlpha())
     }
 
+    @Test
+    fun testTinting() {
+        try {
```
Comment (adam-arold) [inline zircon.core/src/jvmTest/kotlin/org/hexworks/zircon/internal/color/DefaultTileColorTest.kt]: You can use `assertFailsWith` here instead of the `try`/`catch`:

```

@Test
fun `calling hereIsAnException will return an exception no matter what`() {
  assertFailsWith<IllegalArgumentException> { hereIsAnException() }
}
```

**Comment by klaushauschild1984** [comment]
@adam-arold I recently found `ColorInterpolator` which is capable of interpolate one color into another. From a mathematical point of view is tint, shade and tone interpolation of the given color to white, black, and grey. The target grey could be calculated by arithmetic middle of R, G and B. Besides edge cases this should simplify implementation in `TileColor`. The interpolation code is not that expensive too. What do you think?

**Comment by adam-arold** [comment]
This is a great idea! Besides if we end up changing how interpolation works we'll only have to touch it once. Great find! 👍

**Comment by klaushauschild1984** [comment]
My last commit changed implementation of `tint` and `shade` to use also interpolation. My recently introduced tests needed alignment. That means API behavior was changed. Not sure how to handle this. Do you wanna keep the original implementation and treat it as "correct" or do you accept this change?

**Comment by adam-arold** [comment]
Would you like to join our [Discord Server](https://discordapp.com/invite/vSNgvBh) to talk about this? Then we'd have the chance to talk this through with the others involved in this topic.
